### DISCUSSION TITLE
Crowdsec for use with hostsystem

### DISCUSSION THREAD

**Discussion by gomaaz** [ZoeyVid/NPMplus#1299] [Q&A] [answered]
Hello,

I would like to use CrowdSec not only with NPMplus but also for my host system, and I want to add an additional firewall bouncer. However, the bouncers can only be installed using apt. CrowdSec itself is installed via the apt package manager, which is not available in Docker containers.

Thank you for your help!

**Comment by Zoey2936** [comment, marked as answer]
hello, yes this works, you can either install crowdsec via apt and install the NPMplus collection, then you can use the same guide as you see in this readme but the commands need to be run on yopur system so without docker exec or you install it via docker and connect the firewall bouncer to the docker container

**Reply by gomaaz** [reply]
thanks

**Comment by gomaaz** [comment]
hmm

I just installed two bouncers. 
- the npmplus from this github page with the given yaml and instructions and
- the firewall bouncer for iptables

I can't view the npmplus bouncer here, is this normal?

by manually test, I achieved blocking my ip for 2 Minutes. It also works on docker (npmplus) proxy hosts.

The logs of /opt/npm/nginx/access.log are not showing up

I wondering if the npm bouncer is running...

![image](https://github.com/user-attachments/assets/1a509d56-a779-42e9-b9e2-425e6f781003)
![image](https://github.com/user-attachments/assets/9c7276d4-d909-49e8-a993-393ced3c696f)

**Reply by Zoey2936** [reply]
where did you put the npmplus.yaml?

**Comment by gomaaz** [comment]
also the npmplus.yaml contains source: docker entries. Is this still relevant for crowdsec natively installed on the host system?

**Reply by Zoey2936** [reply]
yes, crowdsec needs to read npmplus docker logs

**Comment by gomaaz** [comment]
ok....

I forgot to restart the npmplus container...

Now its working and I am immediately blocked for unknown reason 😂

Only nginx-access with all its proxies is blocked, everything else like ssh works... 
what is causing this?

**Reply by Zoey2936** [reply]
what is written in your docker logs?

**Reply by gomaaz** [reply]
This is the npmplus docker log last 50 Lines
I obfuscated ips and domains

Crowdsec access forbidden is shown on all proxies

> 2024/12/11 10:35:06 [error] 1069#1069: *173 [lua] crowdsec.lua:651: Allow(): AppSec check: connection refused, client: 1.2.3.4, server: mydomain.com, request: "GET /web/1.2.3.4HTTP/2.0", host: "mydomain.com"
> 2024/12/11 10:35:06 [alert] 1069#1069: *173 [lua] crowdsec.lua:718: Allow(): [Crowdsec] denied '1.2.3.4' with 'ban' (by appsec), client: 1.2.3.4, server: mydomain.com, request: "GET /web/1.2.3.4HTTP/2.0", host: "mydomain.com"
> 2024/12/11 10:35:06 [error] 1069#1069: *173 connect() failed (111: Connection refused), client: 1.2.3.4, server: mydomain.com, request: "GET /1.2.3.4HTTP/2.0", host: "mydomain.com", referrer: "https://mydomain.com/web/index.html"
> 2024/12/11 10:35:06 [error] 1069#1069: *173 [lua] crowdsec.lua:578: AppSecCheck(): Fallback because of err: connection refused, client: 1.2.3.4, server: mydomain.com, request: "GET /1.2.3.4HTTP/2.0", host: "mydomain.com", referrer: "https://mydomain.com/web/index.html"
> 2024/12/11 10:35:06 [error] 1069#1069: *173 [lua] crowdsec.lua:651: Allow(): AppSec check: connection refused, client: 1.2.3.4, server: mydomain.com, request: "GET /1.2.3.4HTTP/2.0", host: "mydomain.com", referrer: "https://mydomain.com/web/index.html"
> 2024/12/11 10:35:06 [alert] 1069#1069: *173 [lua] crowdsec.lua:718: Allow(): [Crowdsec] denied '1.2.3.4' with 'ban' (by appsec), client: 1.2.3.4, server: mydomain.com, request: "GET /1.2.3.4HTTP/2.0", host: "mydomain.com", referrer: "https://mydomain.com/web/index.html"
> 2024/12/11 10:35:06 [error] 1069#1069: *173 connect() failed (111: Connection refused), client: 1.2.3.4, server: mydomain.com, request: "GET /web/1.2.3.4HTTP/2.0", host: "mydomain.com"
> 2024/12/11 10:35:06 [error] 1069#1069: *173 [lua] crowdsec.lua:578: AppSecCheck(): Fallback because of err: connection refused, client: 1.2.3.4, server: mydomain.com, request: "GET /web/1.2.3.4HTTP/2.0", host: "mydomain.com"
> 2024/12/11 10:35:06 [error] 1069#1069: *173 [lua] crowdsec.lua:651: Allow(): AppSec check: connection refused, client: 1.2.3.4, server: mydomain.com, request: "GET /web/1.2.3.4HTTP/2.0", host: "mydomain.com"
> 2024/12/11 10:35:06 [alert] 1069#1069: *173 [lua] crowdsec.lua:718: Allow(): [Crowdsec] denied '1.2.3.4' with 'ban' (by appsec), client: 1.2.3.4, server: mydomain.com, request: "GET /web/1.2.3.4HTTP/2.0", host: "mydomain.com"
> 2024/12/11 10:35:06 [error] 1069#1069: *173 connect() failed (111: Connection refused), client: 1.2.3.4, server: mydomain.com, request: "GET /1.2.3.4HTTP/2.0", host: "mydomain.com", referrer: "https://mydomain.com/web/index.html"
> 2024/12/11 10:35:06 [error] 1069#1069: *173 [lua] crowdsec.lua:578: AppSecCheck(): Fallback because of err: connection refused, client: 1.2.3.4, server: mydomain.com, request: "GET /1.2.3.4HTTP/2.0", host: "mydomain.com", referrer: "https://mydomain.com/web/index.html"
> 2024/12/11 10:35:06 [error] 1069#1069: *173 [lua] crowdsec.lua:651: Allow(): AppSec check: connection refused, client: 1.2.3.4, server: mydomain.com, request: "GET /1.2.3.4HTTP/2.0", host: "mydomain.com", referrer: "https://mydomain.com/web/index.html"
> 2024/12/11 10:35:06 [alert] 1069#1069: *173 [lua] crowdsec.lua:718: Allow(): [Crowdsec] denied '1.2.3.4' with 'ban' (by appsec), client: 1.2.3.4, server: mydomain.com, request: "GET /1.2.3.4HTTP/2.0", host: "mydomain.com", referrer: "https://mydomain.com/web/index.html"
> 2024/12/11 10:35:08 [error] 1069#1069: *173 connect() failed (111: Connection refused), client: 1.2.3.4, server: mydomain.com, request: "GET /web/1.2.3.4HTTP/2.0", host: "mydomain.com", referrer: "https://mydomain.com/web/serviceworker.js"
> 2024/12/11 10:35:08 [error] 1069#1069: *173 [lua] crowdsec.lua:578: AppSecCheck(): Fallback because of err: connection refused, client: 1.2.3.4, server: mydomain.com, request: "GET /web/1.2.3.4HTTP/2.0", host: "mydomain.com", referrer: "https://mydomain.com/web/serviceworker.js"
> 2024/12/11 10:35:08 [error] 1069#1069: *173 [lua] crowdsec.lua:651: Allow(): AppSec check: connection refused, client: 1.2.3.4, server: mydomain.com, request: "GET /web/1.2.3.4HTTP/2.0", host: "mydomain.com", referrer: "https://mydomain.com/web/serviceworker.js"
> 2024/12/11 10:35:08 [alert] 1069#1069: *173 [lua] crowdsec.lua:718: Allow(): [Crowdsec] denied '1.2.3.4' with 'ban' (by appsec), client: 1.2.3.4, server: mydomain.com, request: "GET /web/1.2.3.4HTTP/2.0", host: "mydomain.com", referrer: "https://mydomain.com/web/serviceworker.js"
> 2024/12/11 10:35:08 [error] 1069#1069: *173 connect() failed (111: Connection refused), client: 1.2.3.4, server: mydomain.com, request: "GET /web/1.2.3.4HTTP/2.0", host: "mydomain.com"
> 2024/12/11 10:35:08 [error] 1069#1069: *173 [lua] crowdsec.lua:578: AppSecCheck(): Fallback because of err: connection refused, client: 1.2.3.4, server: mydomain.com, request: "GET /web/1.2.3.4HTTP/2.0", host: "mydomain.com"
> 2024/12/11 10:35:08 [error] 1069#1069: *173 [lua] crowdsec.lua:651: Allow(): AppSec check: connection refused, client: 1.2.3.4, server: mydomain.com, request: "GET /web/1.2.3.4HTTP/2.0", host: "mydomain.com"
> 2024/12/11 10:35:08 [alert] 1069#1069: *173 [lua] crowdsec.lua:718: Allow(): [Crowdsec] denied '1.2.3.4' with 'ban' (by appsec), client: 1.2.3.4, server: mydomain.com, request: "GET /web/1.2.3.4HTTP/2.0", host: "mydomain.com"
> 2024/12/11 10:35:08 [error] 1069#1069: *173 connect() failed (111: Connection refused), client: 1.2.3.4, server: mydomain.com, request: "GET /1.2.3.4HTTP/2.0", host: "mydomain.com", referrer: "https://mydomain.com/web/index.html"
> 2024/12/11 10:35:08 [error] 1069#1069: *173 [lua] crowdsec.lua:578: AppSecCheck(): Fallback because of err: connection refused, client: 1.2.3.4, server: mydomain.com, request: "GET /1.2.3.4HTTP/2.0", host: "mydomain.com", referrer: "https://mydomain.com/web/index.html"
> 2024/12/11 10:35:08 [error] 1069#1069: *173 [lua] crowdsec.lua:651: Allow(): AppSec check: connection refused, client: 1.2.3.4, server: mydomain.com, request: "GET /1.2.3.4HTTP/2.0", host: "mydomain.com", referrer: "https://mydomain.com/web/index.html"
> 2024/12/11 10:35:08 [alert] 1069#1069: *173 [lua] crowdsec.lua:718: Allow(): [Crowdsec] denied '1.2.3.4' with 'ban' (by appsec), client: 1.2.3.4, server: mydomain.com, request: "GET /1.2.3.4HTTP/2.0", host: "mydomain.com", referrer: "https://mydomain.com/web/index.html"
> 2024/12/11 10:35:09 [error] 1069#1069: *173 connect() failed (111: Connection refused), client: 1.2.3.4, server: mydomain.com, request: "GET /web/1.2.3.4HTTP/2.0", host: "mydomain.com"
> 2024/12/11 10:35:09 [error] 1069#1069: *173 [lua] crowdsec.lua:578: AppSecCheck(): Fallback because of err: connection refused, client: 1.2.3.4, server: mydomain.com, request: "GET /web/1.2.3.4HTTP/2.0", host: "mydomain.com"
> 2024/12/11 10:35:09 [error] 1069#1069: *173 [lua] crowdsec.lua:651: Allow(): AppSec check: connection refused, client: 1.2.3.4, server: mydomain.com, request: "GET /web/1.2.3.4HTTP/2.0", host: "mydomain.com"
> 2024/12/11 10:35:09 [alert] 1069#1069: *173 [lua] crowdsec.lua:718: Allow(): [Crowdsec] denied '1.2.3.4' with 'ban' (by appsec), client: 1.2.3.4, server: mydomain.com, request: "GET /web/1.2.3.4HTTP/2.0", host: "mydomain.com"
> 2024/12/11 10:35:09 [error] 1069#1069: *173 connect() failed (111: Connection refused), client: 1.2.3.4, server: mydomain.com, request: "GET /1.2.3.4HTTP/2.0", host: "mydomain.com", referrer: "https://mydomain.com/web/index.html"
> 2024/12/11 10:35:09 [error] 1069#1069: *173 [lua] crowdsec.lua:578: AppSecCheck(): Fallback because of err: connection refused, client: 1.2.3.4, server: mydomain.com, request: "GET /1.2.3.4HTTP/2.0", host: "mydomain.com", referrer: "https://mydomain.com/web/index.html"
> 2024/12/11 10:35:09 [error] 1069#1069: *173 [lua] crowdsec.lua:651: Allow(): AppSec check: connection refused, client: 1.2.3.4, server: mydomain.com, request: "GET /1.2.3.4HTTP/2.0", host: "mydomain.com", referrer: "https://mydomain.com/web/index.html"
> 2024/12/11 10:35:09 [alert] 1069#1069: *173 [lua] crowdsec.lua:718: Allow(): [Crowdsec] denied '1.2.3.4' with 'ban' (by appsec), client: 1.2.3.4, server: mydomain.com, request: "GET /1.2.3.4HTTP/2.0", host: "mydomain.com", referrer: "https://mydomain.com/web/index.html"
> 2024/12/11 10:35:10 [error] 1069#1069: *173 connect() failed (111: Connection refused), client: 1.2.3.4, server: mydomain.com, request: "GET /web/1.2.3.4HTTP/2.0", host: "mydomain.com", referrer: "https://mydomain.com/web/serviceworker.js"
> 2024/12/11 10:35:10 [error] 1069#1069: *173 [lua] crowdsec.lua:578: AppSecCheck(): Fallback because of err: connection refused, client: 1.2.3.4, server: mydomain.com, request: "GET /web/1.2.3.4HTTP/2.0", host: "mydomain.com", referrer: "https://mydomain.com/web/serviceworker.js"
> 2024/12/11 10:35:10 [error] 1069#1069: *173 [lua] crowdsec.lua:651: Allow(): AppSec check: connection refused, client: 1.2.3.4, server: mydomain.com, request: "GET /web/1.2.3.4HTTP/2.0", host: "mydomain.com", referrer: "https://mydomain.com/web/serviceworker.js"
> 2024/12/11 10:35:10 [alert] 1069#1069: *173 [lua] crowdsec.lua:718: Allow(): [Crowdsec] denied '1.2.3.4' with 'ban' (by appsec), client: 1.2.3.4, server: mydomain.com, request: "GET /web/1.2.3.4HTTP/2.0", host: "mydomain.com", referrer: "https://mydomain.com/web/serviceworker.js"
> 2024/12/11 10:35:10 [error] 1070#1070: *200 connect() failed (111: Connection refused), client: 1.2.3.4, server: _, request: "GET /users HTTP/2.0", host: "1.2.3.4:81"
> 2024/12/11 10:35:10 [error] 1070#1070: *200 [lua] crowdsec.lua:578: AppSecCheck(): Fallback because of err: connection refused, client: 1.2.3.4, server: _, request: "GET /users HTTP/2.0", host: "1.2.3.4:81"
> 2024/12/11 10:35:10 [error] 1070#1070: *200 [lua] crowdsec.lua:651: Allow(): AppSec check: connection refused, client: 1.2.3.4, server: _, request: "GET /users HTTP/2.0", host: "1.2.3.4:81"
> 2024/12/11 10:35:10 [alert] 1070#1070: *200 [lua] crowdsec.lua:718: Allow(): [Crowdsec] denied '1.2.3.4' with 'ban' (by appsec), client: 1.2.3.4, server: _, request: "GET /users HTTP/2.0", host: "1.2.3.4:81"
> 2024/12/11 10:35:10 [error] 1070#1070: *200 connect() failed (111: Connection refused), client: 1.2.3.4, server: _, request: "GET /1.2.3.4HTTP/2.0", host: "1.2.3.4:81", referrer: "https://1.2.3.4:81/users"
> 2024/12/11 10:35:10 [error] 1070#1070: *200 [lua] crowdsec.lua:578: AppSecCheck(): Fallback because of err: connection refused, client: 1.2.3.4, server: _, request: "GET /1.2.3.4HTTP/2.0", host: "1.2.3.4:81", referrer: "https://1.2.3.4:81/users"
> 2024/12/11 10:35:10 [error] 1070#1070: *200 [lua] crowdsec.lua:651: Allow(): AppSec check: connection refused, client: 1.2.3.4, server: _, request: "GET /1.2.3.4HTTP/2.0", host: "1.2.3.4:81", referrer: "https://1.2.3.4:81/users"
> 2024/12/11 10:35:10 [alert] 1070#1070: *200 [lua] crowdsec.lua:718: Allow(): [Crowdsec] denied '1.2.3.4' with 'ban' (by appsec), client: 1.2.3.4, server: _, request: "GET /1.2.3.4HTTP/2.0", host: "1.2.3.4:81", referrer: "https://1.2.3.4:81/users"
> 2024/12/11 10:35:11 [error] 1069#1069: *173 connect() failed (111: Connection refused), client: 1.2.3.4, server: mydomain.com, request: "GET /web/1.2.3.4HTTP/2.0", host: "mydomain.com", referrer: "https://mydomain.com/web/serviceworker.js"
> 2024/12/11 10:35:11 [error] 1069#1069: *173 [lua] crowdsec.lua:578: AppSecCheck(): Fallback because of err: connection refused, client: 1.2.3.4, server: mydomain.com, request: "GET /web/1.2.3.4HTTP/2.0", host: "mydomain.com", referrer: "https://mydomain.com/web/serviceworker.js"
> 2024/12/11 10:35:11 [error] 1069#1069: *173 [lua] crowdsec.lua:651: Allow(): AppSec check: connection refused, client: 1.2.3.4, server: mydomain.com, request: "GET /web/1.2.3.4HTTP/2.0", host: "mydomain.com", referrer: "https://mydomain.com/web/serviceworker.js"
> 2024/12/11 10:35:11 [alert] 1069#1069: *173 [lua] crowdsec.lua:718: Allow(): [Crowdsec] denied '1.2.3.4' with 'ban' (by appsec), client: 1.2.3.4, server: mydomain.com, request: "GET /web/1.2.3.4HTTP/2.0", host: "mydomain.com", referrer: "https://mydomain.com/web/serviceworker.js"

**Reply by gomaaz** [reply]
I also tried other source IPs from my VPN Providers. 
Still blocked... it seems like everything is blocked.

**Reply by Zoey2936** [reply]
where did you put the npmplus.yaml? and did you installed the NPMplus collection?

**Reply by gomaaz** [reply]
right here: /opt/crowdsec/conf/acquis.d/npmplus.yaml

yes I isntalled it right from bouncer hub

**Reply by Zoey2936** [reply]
you need to adjust the path to the path crowdsec uses when installed on the host, it should be somewhere in /etc/crowdsec, there should be somewhere the acquis.d folder in which the npmplus.yaml should be placed

**Reply by gomaaz** [reply]
ok this looks better now:

I moved 
/opt/crowdsec/conf/acquis.d/npmplus.yaml
to 
/etc/crowdsec/acquis.d/npmplus.yaml

Got access again on proxies
log paths are also shown now:

![image](https://github.com/user-attachments/assets/b6604bb6-f491-41fd-8bd9-14a232340366)
![image](https://github.com/user-attachments/assets/6e71c653-be8a-4c19-a67d-462926be6255)

I guess its fixed then.

I recommend to mention this in your install guide?...

**Reply by Zoey2936** [reply]
the guide is written for docker install, which would also work in your case, since the firewall bouncer, even installed on host, can connect to the crowdsec api when running in docker, I don't see a reason to explain a more complicated install

**Reply by gomaaz** [reply]
anyway, thanks for your support 🙏

**Comment by DJKatastrof** [comment]
I seem to have the same issue, 
My docker yaml,
Any ideas? 
```
  crowdsec:
    container_name: crowdsec
    image: docker.io/crowdsecurity/crowdsec:latest
    restart: always
    network_mode: bridge
    ports:
      - 127.0.0.1:7422:7422
      - 127.0.0.1:8080:8080
    environment:
      - TZ=Europe/Stockholm # needs to be changed
      - COLLECTIONS=ZoeyVid/npmplus
    volumes:
      - ./npmplus/crowdsec/conf:/etc/crowdsec
      - ./npmplus/crowdsec/data:/var/lib/crowdsec/data
      - ./npmplus/nginx:/opt/npmplus/nginx:ro
      - /var/run/docker.sock:/var/run/docker.sock:ro
```
![image](https://github.com/user-attachments/assets/70128b8f-8dca-4101-8d30-45388251fa20)

![image](https://github.com/user-attachments/assets/2829b85e-e2cd-4fc6-944c-1c73e6bec440)

**Reply by Zoey2936** [reply]
looks like you have a diffrent issue, but please send your full compose.yaml

**Reply by DJKatastrof** [reply]
Want me to create a new thread? or keep going here?

**Reply by Zoey2936** [reply]
new thread would be good
